src/sidebar/components/test/annotation-user-test.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const { createElement } = require('preact');
+const { shallow } = require('enzyme');
+
+const AnnotationUser = require('../annotation-user');
+
+describe('AnnotationUser', () => {
+  let fakeAnnotation;
+  let fakeFeatures;
+  let fakeIsThirdPartyUser;
+  let fakeServiceUrl;
+  let fakeSettings;
+  let fakeUsername;
+
+  const createAnnotationUser = () => {
+    return shallow(
+      <AnnotationUser
+        annotation={fakeAnnotation}
+        features={fakeFeatures}
+        serviceUrl={fakeServiceUrl}
+        settings={fakeSettings}
+      />
+    ).dive(); // Dive needed because of `withServices` usage in component
+  };
+
+  beforeEach(() => {
+    fakeAnnotation = {
+      user: {
+        userid: 'someone@hypothes.is',
+      },
+    };
+    fakeFeatures = { flagEnabled: sinon.stub() };
+    fakeIsThirdPartyUser = sinon.stub();
+    fakeServiceUrl = sinon.stub();
+    fakeSettings = {};
+    fakeUsername = sinon.stub();
+
+    AnnotationUser.$imports.$mock({
+      '../util/account-id': {
+        isThirdPartyUser: fakeIsThirdPartyUser,
+        username: fakeUsername,
+      },
+    });
+  });
+
+  afterEach(() => {
+    AnnotationUser.$imports.$restore();
+  });
+
+  describe('link to user activity', () => {
+    context('first-party user', () => {
+      it('should provide a link to the user profile', () => {
+        fakeIsThirdPartyUser.returns(false);
+        fakeServiceUrl.returns('link-to-user');
+
+        const wrapper = createAnnotationUser();
+        const linkEl = wrapper.find('a');
+
+        assert.isOk(linkEl.exists());
+        assert.calledWith(fakeServiceUrl, 'user', {
+          user: fakeAnnotation.user,
+        });
+        assert.equal(linkEl.prop('href'), 'link-to-user');
+      });
+    });
+
+    context('third-party user', () => {
+      beforeEach(() => {
+        fakeIsThirdPartyUser.returns(true);
+      });
+
+      it('should link to user if `settings.usernameUrl` is set', () => {
+        fakeSettings.usernameUrl = 'http://example.com?user=';
+        fakeUsername.returns('elephant');
+
+        const wrapper = createAnnotationUser();
+        const linkEl = wrapper.find('a');
+
+        assert.isOk(linkEl.exists());
+        assert.equal(linkEl.prop('href'), 'http://example.com?user=elephant');
+      });
+
+      it('should not link to user if `settings.usernameUrl` is not set', () => {
+        const wrapper = createAnnotationUser();
+        const linkEl = wrapper.find('a');
+
+        assert.isNotOk(linkEl.exists());
+      });
+    });
+  });
+
+  describe('rendered user name', () => {
+    context('feature flag on', () => {
+      beforeEach(() => {
+        fakeFeatures.flagEnabled.returns(true);
```

Comment:
This would return true even if `flagEnabled` was called with the wrong flag. You can make the test more specific using the `stub.withArgs(<args>).returns(<return value>)` API.